src/framework-configuration.js

```diff
@@ -223,19 +261,28 @@ export class FrameworkConfiguration {
 
     for (let i = 0, ii = toAdd.length; i < ii; ++i) {
       resource = toAdd[i];
-      if (typeof resource !== 'string') {
+      if (!resource) {
```

Comment:
The way `resource` type is checked doesn't make sense to me here?
For example, if an `object` is passed, it will go through the `else` branch below (supposed to be a function) rather than here, right?

Here, I am expecting something like:
```
if (typeof resource === 'string') ...
else if (typeof resource === 'function') ...
else throw new Error(`Invalid resource [${resource}]. Resources must be specified as functions or relative module IDs.`).
```
Note that I tweaked the error message slightly to account for the new type.

Reply:
Yes, not sure what I was thinking here. Probably mainly because I was hoping the typescript definition will do the job instead of real checks.